Text index: disable splitting index data into multiple segments by default

ahmadov opened the pull request:
<!---
A technical comment, you are free to remove or leave it as it is when PR is created
The following categories are used in the next scripts, update them accordingly
utils/changelog/changelog.py
tests/ci/cancel_and_rerun_workflow_lambda/app.py
-->
### Changelog category (leave one):

- Performance Improvement

### Changelog entry (a [user-readable short description](https://github.com/ClickHouse/ClickHouse/blob/master/docs/changelog_entry_guidelines.md) of the changes that goes into CHANGELOG.md):

Previously, the text index data would be separated into multiple segments (each segment size by default was 256 MiB). This might reduce the memory consumption while building the text index, however this increases the space requirement on the disk and increase the query response time.

With this PR, by default we disable that feature and make the single segment size unlimited, but still allow users to adjust if it is necessary on an environment with a limited memory.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

<!---
Directly edit documentation source files in the "docs" folder with the same pull-request as code changes

or

Add a user-readable short description of the changes that should be added to docs.clickhouse.com below.

At a minimum, the following information should be added (but add more as needed).
- Motivation: Why is this function, table engine, etc. useful to ClickHouse users?

- Parameters: If the feature being added takes arguments, options or is influenced by settings, please list them below with a brief explanation.

- Example use: A query or command.
-->

ahmadov committed:
Text index: disable splitting index data into multiple segments by default

Previously, the text index data would be separated into multiple
segments (each segment size by default was 256 MiB). This might reduce
the memory consumption while building the text index, however this might
increase the space requirement on the disk and increase the query
response time.

With this PR, by default we disable the that feature and make the
segment size unlimited, but still allow users to adjust if it is
necessary on the environment with the limited memory.

ahmadov committed:
add missing merge tree change history entry

Ergus reviewed (commented):
LGTM

rschu1ze commented on docs/en/engines/table-engines/mergetree-family/invertedindexes.md:
```diff
@@ -99,7 +99,7 @@ SELECT * from tab WHERE hasToken(str, 'Hello');
 Furthermore, each text index is internally divided into "segments".
 The existence and size of the segments are generally transparent to users but the segment size determines the memory consumption during index construction (e.g. when two parts are merged).
-Configuration parameter `max_digestion_size_per_segment` (default: 256 MB) controls the amount of data read from the underlying column before a new segment is created.
+Configuration parameter `max_digestion_size_per_segment` (default: unlimited) controls the amount of data read from the underlying column before a new segment is created.
```
`default: 0 which means unlimited, i.e. a single segment is created per index granule`.

rschu1ze replied:
The inverted index docs actually still talks quite enthusiastically about segments on l. 99-104.

Could we hide or de-emphasize that by default using the same trick as [here](https://clickhouse.com/docs/engines/table-engines/mergetree-family/collapsingmergetree)? It should be clear that this is a tool for advanced usage only.

ahmadov commented on src/Storages/MergeTree/MergeTreeSettings.cpp:
```diff
@@ -65,7 +65,7 @@ namespace ErrorCodes
     correspond to one primary key value.
     )", 0) \
-    DECLARE(UInt64, max_digestion_size_per_segment, 256_MiB, R"(
+    DECLARE(UInt64, max_digestion_size_per_segment, 0, R"(
```
Recently, I saw this settings `Setting::merge_tree_read_split_ranges_into_intersecting_and_non_intersecting_injection_probability`. So, I know the reason for this long name :) 

1. Yes, I'd prefer `text_index_segment_max_digestion_size_in_bytes`, so it can start with `text_index_*`

2. I think the only one reason for that, so users can define multiple text indexes for different columns in the same table. So, then in this case it makes to have it as MT settings, since it's likely the same size for all indexes. But for fine-granularity POV it makes sense to make it text index parameter. It shouldn't be difficult to make it the text index parameter.

ahmadov committed:
Deprecate `max_digestion_size_per_segment`

Move the settings into text index parameters to have a better control.

Update docs to reflect the change.

rschu1ze committed:
Merge remote-tracking branch 'ClickHouse/master' into ahmadov/text-index-remove-separate-segments

rschu1ze committed:
Minor fixups

rschu1ze committed:
Fix style

rschu1ze committed:
Fix FastTest
